bin.src/makeLamDesign.py

```diff
@@ -0,0 +1,116 @@
+
+# Mapping of colors to fiberIds
+# Constructed from a snippet by Fabrice Madec, "dummy cable B fibers"
+# https://sumire-pfs.slack.com/files/U3MLENNHH/FFS6P4UR5/dummy_cable_b_fibers.txt
+FIBER_COLORS = {"red1": [2],
+                "red2": [3],
+                "red3": [308],
+                "red4": [339],
+                "red5": [340],
+                "red6": [342],
+                "red7": [649],
+                "red8": [650],
+                "red": [2, 3, 308, 339, 340, 342, 649, 650],
+                "orange": [12, 60, 110, 161, 210, 259, 341],
+                "blue": [32, 111, 223, 289, 418, 518, 620],
+                "green": [63, 192, 255, 401, 464, 525, 587],
+                "yellow": [347, 400, 449, 545, 593, 641],
+                }
+
+# Mapping of colors to hash value
+# This scheme makes the hash look like binary, with a 1 if the color was used and 0 if not
+HASH_COLORS = {color: 16**ii for ii, color in enumerate(sorted(FIBER_COLORS.keys()))}
+
+
+def makePfiDesign(pfiDesignId, fiberId):
+    """Build a ``PfsConfig``
+
+    Parameters
+    ----------
+    pfiDesignId : `int`
+        Identifier for the top-end design. For our purposes, this is just a
+        unique integer.
+    fiberId : `numpy.ndarray` of `int`
+        Array of identifiers for fibers that will be lit.
+
+    Returns
+    -------
+    config : `pfs.datamodel.PfsConfig`
+        Configuration of the top-end.
+    """
+    raBoresight = 0.0
+    decBoresight = 0.0
+    tract = np.zeros_like(fiberId, dtype=int)
+    patch = ["0,0" for _ in fiberId]
+
+    num = len(fiberId)
+    catId = np.zeros_like(fiberId, dtype=int)
+    objId = fiberId
+    targetTypes = TargetType.SCIENCE*np.ones_like(fiberId, dtype=int)
+    ra = np.zeros_like(fiberId, dtype=float)
+    dec = np.zeros_like(fiberId, dtype=float)
+    pfiNominal = np.zeros((num, 2), dtype=float)
+
+    fiberMags = [[] for _ in fiberId]
+    filterNames = [[] for _ in fiberId]
+
+    return PfiDesign(pfiDesignId, raBoresight, decBoresight,
+                     fiberId, tract, patch, ra, dec, catId, objId, targetTypes,
+                     fiberMags, filterNames, pfiNominal)
+
+
+def colorsToFibers(colors):
+    """Convert a list of colors to an array of fiber IDs
+
+    Parameters
+    ----------
+    colors : iterable of `str`
+        List of colors.
+
+    Returns
+    -------
+    fiberId : `numpy.ndarray`
+        Array of fiber IDs.
+    """
+    return np.array(sorted(set(sum([FIBER_COLORS[col] for col in colors], []))))
+
+
+def hashColors(colors):
+    """Convert a list of colors to a hash for the pfiDesignId
+
+    Parameters
+    ----------
+    colors : iterable of `str`
+        List of colors.
+
+    Returns
+    -------
+    hash : `int`
+        Hash, for the pfiDesignId.
+    """
+    return sum(HASH_COLORS[col] for col in set(colors))
+
+
+if __name__ == "__main__":
+    import argparse
+    parser = argparse.ArgumentParser(description="Create a PfiDesign for a LAM exposure, "
```

Comment:
I think we should explain what "colors" [sic] of fibres means.  It's very LAM specific, and while I understand that this is LAM specific code it's in the PFS codebase